doc/cache-spec.md

```diff
@@ -0,0 +1,132 @@
+# Spec for Koop cache modules
+In order to function as a cache for Koop, a module must support the following API
+
+## Functions
+
+### insert
+Insert a new table into the cache
+```javascript
+function (table, data, layer, function (err, success)
+ ```
+* Table (string): the name of the table to be inserted. Usually of the form: type:key
+* Data (object): geojson to be inserted after creation
+* Layer (integer): the id of the layer. Usually defaults to 0
+* Callback (function): returns err on failure or true on success
+
+### insertPartial
+Insert a set of rows into an existing table
+```javascript
+function (table, data, layerId, function (err, success)
+```
+* Table (string): the name of the table to be inserted. Usually of the form: type:key
+* Data (object): geojson to be inserted after creation
+* Layer (integer): the id of the layer. Usually defaults to 0
+* Callback (function): returns err on failure or true on success
+
+### remove
+```javascript
+function (table, function(err, success))
+```
+* Table: the name of the table to be deleted. Includes the layer id 
+ * e.g. Socrata:ejk5-56ko:0
+* Callback: returns err on failure or true on success
```

Comment:
The standard way to deal with callbacks in node is `function (err, params*)`, where `err` is either an [error object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) or `null`. If there's an error nothing else should be passed. If `err` is `null` then response data should be passed afterwards (there's no limit to number of params but it's usually best to limit it to one unless really necessary).

This article gives a good overview of the convention: http://thenodeway.io/posts/understanding-error-first-callbacks/